lib/JsonpMainTemplatePlugin.js

```diff
@@ -2,205 +2,192 @@
-					return "\" + " + this.renderCurrentHashCode(hash, length) + " + \"";
-				}.bind(this),
-				chunk: {
-					id: "\" + chunkId + \"",
-					hash: "\" + " + JSON.stringify(chunkMaps.hash) + "[chunkId] + \"",
-					hashWithLength: function(length) {
-						var shortChunkHashMap = {};
-						Object.keys(chunkMaps.hash).forEach(function(chunkId) {
-							if(typeof chunkMaps.hash[chunkId] === "string")
-								shortChunkHashMap[chunkId] = chunkMaps.hash[chunkId].substr(0, length);
-						});
-						return "\" + " + JSON.stringify(shortChunkHashMap) + "[chunkId] + \"";
-					},
-					name: "\" + (" + JSON.stringify(chunkMaps.name) + "[chunkId]||chunkId) + \""
-				}
-			}) + ";",
-			"var timeout = setTimeout(onScriptComplete, " + chunkLoadTimeout + ");",
-			"script.onerror = script.onload = onScriptComplete;",
-			"function onScriptComplete() {",
-			this.indent([
-				"// avoid mem leaks in IE.",
-				"script.onerror = script.onload = null;",
-				"clearTimeout(timeout);",
-				"var chunk = installedChunks[chunkId];",
-				"if(chunk !== 0) {",
+		});
+		mainTemplate.plugin("require-ensure", function(_, chunk, hash) {
+			return this.asString([
+				"if(installedChunks[chunkId] === 0)",
+				this.indent([
+					"return Promise.resolve();"
+				]),
+				"",
+				"// a Promise means \"currently loading\".",
+				"if(installedChunks[chunkId]) {",
 				this.indent([
-					"if(chunk) chunk[1](new Error('Loading chunk ' + chunkId + ' failed.'));",
-					"installedChunks[chunkId] = undefined;"
+					"return installedChunks[chunkId][2];"
 				]),
-				"}"
-			]),
-			"};",
-		]);
-	});
-	mainTemplate.plugin("require-ensure", function(_, chunk, hash) {
-		return this.asString([
-			"if(installedChunks[chunkId] === 0)",
-			this.indent([
-				"return Promise.resolve();"
-			]),
-			"",
-			"// a Promise means \"currently loading\".",
-			"if(installedChunks[chunkId]) {",
-			this.indent([
-				"return installedChunks[chunkId][2];"
-			]),
-			"}",
-			"// start chunk loading",
-			"var head = document.getElementsByTagName('head')[0];",
-			this.applyPluginsWaterfall("jsonp-script", "", chunk, hash),
-			"",
-			"var promise = new Promise(function(resolve, reject) {",
-			this.indent([
-				"installedChunks[chunkId] = [resolve, reject];"
-			]),
-			"});",
-			"installedChunks[chunkId][2] = promise;",
-			"",
-			"head.appendChild(script);",
-			"return promise;"
-		]);
-	});
-	mainTemplate.plugin("require-extensions", function(source, chunk) {
-		if(chunk.chunks.length === 0) return source;
-		return this.asString([
-			source,
-			"",
-			"// on error function for async loading",
-			this.requireFn + ".oe = function(err) { console.error(err); throw err; };"
-		]);
-	});
-	mainTemplate.plugin("bootstrap", function(source, chunk, hash) {
-		if(chunk.chunks.length > 0) {
-			var jsonpFunction = this.outputOptions.jsonpFunction;
+				"}",
+				"// start chunk loading",
+				"var head = document.getElementsByTagName('head')[0];",
```

Comment:
(unrelated, just a question) `document.head`? Or would there be web compatibility issues?